### PR TITLE
PCGrad gradient surgery for vol_pressure OOD conflict reduction

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

PCGrad (gradient surgery) targets the volume_pressure task's gradient conflict with surface+tau tasks. The 4 OOD geometries (run_133, run_226, run_203, run_158) account for ~92% of squared volume error and produce large, opposing gradients compared to surface pressure. PCGrad projects each task's gradient onto the normal plane of any task whose gradient it conflicts with (negative cosine similarity), removing destructive interference without touching the optimization objective. This should allow the volume_pressure decoder to converge more cleanly on OOD geometry representations.

Reference: Yu et al. 2020, "Gradient Surgery for Multi-Task Learning" (https://arxiv.org/abs/2001.11385)

The core idea: in each backward pass, for each pair of task loss gradients (g_i, g_j), if cos(g_i, g_j) < 0, subtract the conflicting component: g_i ← g_i − (g_i·g_j / ‖g_j‖²)·g_j. This zeroes out the destructive portion without amplifying gradients.

## Instructions

Implement PCGrad for 5 tasks (surface_pressure, tau_x, tau_y, tau_z, volume_pressure) in `train.py`. The implementation should:

1. **Add a `PCGradOptimizer` wrapper class** that wraps the Lion optimizer. After computing per-task losses and gradients separately, it:
   - Stores the gradient vector (flattened shared backbone params) for each task
   - For each pair (i,j) checks `cos_sim = dot(g_i, g_j)` — if < 0, sets `g_i -= dot(g_i, g_j) / ||g_j||^2 * g_j`
   - Sums the projected gradients and writes back to param.grad
   - Then calls the underlying Lion step

2. **Compute per-task losses separately** to get per-task gradients:
   - `loss_sp = L2(pred_surf_pressure, gt_surf_pressure)`
   - `loss_tx = L2(pred_tau_x, gt_tau_x)`, `loss_ty = L2(pred_tau_y, gt_tau_y)`, `loss_tz = L2(pred_tau_z, gt_tau_z)`
   - `loss_vp = L2(pred_vol_pressure, gt_vol_pressure)`
   - Apply the manual loss weights: sp×2.0, tx×1.0, ty×1.5, tz×2.0, vp×(--volume-loss-weight default 1.0)

3. **Gate on `--use-pcgrad` flag** (new boolean argparse flag). When disabled, falls back to the existing combined-loss backward path.

4. **Apply PCGrad to shared backbone parameters only** — skip the surface-specific and volume-specific head parameters (they receive gradients from only one task anyway).

Run with full SOTA config + `--use-pcgrad`:

```bash
cd target/ && torchrun --standalone --nproc_per_node=8 train.py \
  --agent askeladd --optimizer lion --lr 9e-5 --weight-decay 5e-4 \
  --tau-y-loss-weight 1.5 --tau-z-loss-weight 2.0 --surface-loss-weight 2.0 \
  --ema-decay 0.999 --grad-clip-norm 0.5 --lr-warmup-epochs 1 \
  --pos-encoding-mode string_separable --use-qk-norm \
  --rff-num-features 16 --rff-init-sigmas "0.25,0.5,1.0,2.0,4.0" \
  --lr-cosine-t-max 13 --epochs 13 \
  --vol-points-schedule "0:16384:3:32768:6:49152:9:65536" \
  --no-compile-model \
  --model-layers 5 --model-hidden-dim 512 --model-heads 4 --model-slices 128 \
  --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --use-surf-to-vol-xattn \
  --use-pcgrad \
  --wandb-group askeladd-pcgrad-surgery \
  --wandb-name askeladd/pcgrad-surgery-run1
```

Watch memory carefully: PCGrad requires 5 separate backward passes through the backbone. If OOM, reduce `--train-volume-points 49152` or `--batch-size 3`. Consider using gradient checkpointing if available.

If the 5-backward approach hits memory limits, implement a fast approximate version: accumulate all 5 task losses into a combined loss, use a single backward, but then apply PCGrad correction on the already-computed param.grad by decomposing it into task components via per-task loss scaling (note: this approximation is less faithful but avoids the 5x memory overhead).

## Baseline (current SOTA to beat)

| Metric | Val | Test |
|---|---:|---:|
| abupt | **6.4407%** | 7.6992% |
| surface_pressure | 4.1836% | 3.8451% |
| volume_pressure | 3.8557% | **11.6704%** |
| wall_shear | 7.3448% | 7.0429% |

**Single-model training gate:** val_abupt < **6.4407%**  
**W&B run (SOTA):** `ghh0s4ne` (PR #823, group `nezuko-surf-vol-xattn`)
**W&B project:** `wandb-applied-ai-team/senpai-v1-drivaerml-ddp8`

**Primary target:** volume_pressure test error — currently 11.67%, target 6.08% (AB-UPT benchmark)

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #903 because ADVISOR: Closing — this experiment duplicates PR #871 (tanjiro PCGrad, CLOSED), which was scientifically falsified. PCGrad's per-task backward pass requires ~2× wall-clock overhead and at EP4 showed no meaningful gradient conflict reduction (conflict rate converged to baseline). More critically, the OOD gap is a generalization problem, not a gradient conflict problem — the 4 OOD test geometries are test-side only and cannot be addressed via backward-pass surgery. askeladd is already assigned to PR #894 (learned surf pool xattn) which remains the active WIP. Closing this PR to avoid confusion..
